internal/pkg/utils/etcdlogger/assert.go

```diff
@@ -27,3 +34,22 @@ func Assert(t assert.TestingT, expected, actual string) bool {
 
 	return wildcards.Assert(t, expected, actual)
 }
+
+// AssertFromFile compares logs captured by the KVLogWrapper.
+// Comments "// ..." and empty lines are ignored.
+func AssertFromFile(t assert.TestingT, path, actual string) bool {
+	result := false
+	data, err := os.ReadFile(path) //nolint:forbidigo // no virtual FS
+	if assert.NoError(t, err) || errors.Is(err, os.ErrNotExist) {
+		expected := string(data)
+		result = Assert(t, expected, actual)
+	}
```

Comment:
I could not find usage within this PR.  The `if` condition  asserts actual vs. expected, when file within given `path` was possible to read or when it does not exists? Otherwise it creates state dump into `.out` directory. 

Is the dumping section usefull? It covers edge case like `permission denied`.

Reply:
> I could not find usage within this PR. 

Yes, you should search for usage in [the final code](https://github.com/keboola/keboola-as-code/pull/1759), (or do you want to review all in one PR? :slightly_smiling_face:  )

--------------------

>  The if condition ...

The if condition contains `errors.Is(err, os.ErrNotExist)` to print actual keys, even if the file doesn't exists.
It also dumps the actual state to the `.out` dir, so you can easily create the expected file, by copying the actual file.
The "file not exist" error is printed at first.

-------------------

> Is the dumping section usefull? It covers edge case like permission denied.

I don't understand.

Reply:
I'll try to adjust it in another thread